pkg/agent/controller/noderoute/node_route_controller_test.go

```diff
@@ -631,3 +615,96 @@ func TestGetPodCIDRsOnNode(t *testing.T) {
 		})
 	}
 }
+
+func TestRemoveStaleGatewayRoutes(t *testing.T) {
+	fakeClientSet := fake.NewSimpleClientset(node1)
+	c := newController(t, &config.NetworkConfig{}, fakeClientSet)
+	defer c.queue.ShutDown()
+
+	stopCh := make(chan struct{})
+	defer close(stopCh)
+	c.informerFactory.Start(stopCh)
+	c.informerFactory.WaitForCacheSync(stopCh)
+
+	c.routeClient.EXPECT().Reconcile([]string{podCIDR.String()})
+	err := c.removeStaleGatewayRoutes()
+	assert.NoError(t, err)
+}
+
+func TestRemoveStaleWireGuardPeers(t *testing.T) {
+	nodeX := &corev1.Node{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:        "nodeX",
+			Annotations: map[string]string{types.NodeWireGuardPublicAnnotationKey: "fakekey"},
+		},
+	}
+	fakeClientSet := fake.NewSimpleClientset(nodeX)
+	c := newController(t, &config.NetworkConfig{
+		TrafficEncryptionMode: config.TrafficEncryptionModeWireGuard,
+	}, fakeClientSet)
+	defer c.queue.ShutDown()
+
+	stopCh := make(chan struct{})
+	defer close(stopCh)
+	c.informerFactory.Start(stopCh)
+	c.informerFactory.WaitForCacheSync(stopCh)
+
+	err := c.removeStaleWireGuardPeers()
+	assert.NoError(t, err)
```

Comment:
what's the expectation here? shouldn't `c.wireGuardClient.RemoveStalePeers` be called?

Reply:
The old mock wireguard client was added manually, no expected call for the mocked methods. Refine the mock client to generate it automatically and added the expected call here.